Libraries/PeriphDrivers/Source/UART/uart_reva.c

```diff
@@ -548,6 +560,49 @@ unsigned int MXC_UART_RevA_ReadRXFIFO(mxc_uart_reva_regs_t *uart, unsigned char
     return read;
 }
 
+#if MXC_DMA_INSTANCES > 1
+
+void MXC_UART_RevA_DMA0_Handler(void)
+{
+    MXC_DMA_Handler(MXC_DMA0);
+}
+
+void MXC_UART_RevA_DMA1_Handler(void)
+{
+    MXC_DMA_Handler(MXC_DMA1);
+}
+
+#endif
+
+/* "Auto" handlers just need to call MXC_DMA_Handler with the correct
+DMA instance.
+*/
+void MXC_UART_RevA_DMA_SetupAutoHandlers(mxc_dma_regs_t *dma_instance, unsigned int channel)
+{
+#ifdef __arm__
+    NVIC_EnableIRQ(MXC_DMA_CH_GET_IRQ(channel));
+
+#if MXC_DMA_INSTANCES > 1
+    /* (JC): This is not the cleanest or most scalable way to do this,
+        but I tried defining default handler's in the system file.
+        Some complications make this the most attractive short-term
+        option.  We could handle multiple DMA instances better in the DMA API (See the mismatch between the size of "dma_resource" array and the number of channels per instance, to start)*/
+    if (dma_instance == MXC_DMA0) {
+        MXC_NVIC_SetVector(MXC_DMA_CH_GET_IRQ(channel), MXC_UART_RevA_DMA0_Handler);
+    } else if (dma_instance == MXC_DMA1) {
+        MXC_NVIC_SetVector(MXC_DMA_CH_GET_IRQ(channel), MXC_UART_RevA_DMA1_Handler);
+    }
+#else
+    // Only one DMA instance, we can point direct to MXC_DMA_Handler
+    MXC_NVIC_SetVector(MXC_UART_RevA_DMA0_Handler, MXC_DMA_Handler);
```

Comment:
Shouldn't this be `MXC_NVIC_SetVector(MXC_DMA_GET_CH_IRQ(channel), MXC_DMA_Handler)` ?

Reply:
https://github.com/Analog-Devices-MSDK/msdk/pull/763/commits/2cd7d4ffd4a40f7489e81f3c398f9cbe0044256c